org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/DocumentProvider.xtend

```diff
@@ -35,10 +39,17 @@ class DocumentProvider {
 
 	def boolean create(String resourcePath, String content) {
 		val file = getWorkspaceFile(resourcePath)
+
 		val created = create(file)
 		if (created) {
-			file.write(content, '''add file: «file.name»''')
+			if (content !== null && !content.empty) {
+				Files.asCharSink(file, UTF_8).write(content)
+			}
+			file.commit('''add file: «file.name»''')
 		}
+
+		repoSync[onConflict|onConflict.resetToRemoteAndCreateBackup(resourcePath, content)]
```

Comment:
I guess, though you could argue that you'd want to sync the repo regardless.

Anyways, there seems to be a more serious oversight here, which is that calling the method in a state where the file to be created already exists is illegal, and should be reported back to the caller. Otherwise, test editor users might be surprised to find content in a file unknown to them, even though they seemingly just created that file themselves.